### PR TITLE
Move `hidden_layout()` to parent scope

### LINKED ISSUE #201: `hidden_layout()` does not need to be a locally defined method
## What problem does this solve or what need does it fill?

`fn hidden_layout()` is currently defined within the scope of `fn compute_preliminary()`

## What solution would you like?

Move it to the main scope of `flexbox.rs`. It should be more easily testable that way.

## Additional context

Code: https://github.com/DioxusLabs/taffy/blob/main/src/flexbox.rs#L1721

### PR DESCRIPTION
**Pull request by Weibye**
# Objective

- Fix #201 
- Added a test

### CODE REVIEW

File: src/layout.rs [hunk 1]
```diff
@@ -17,11 +17,17 @@ pub struct Layout {
...
 
 impl Layout {
-    /// Creates a new [`Layout`] struct with zero size positioned at the origin
+    /// Creates a new [`Layout`] struct with zero size positioned at the origin.
     #[must_use]
     pub const fn new() -> Self {
         Self { order: 0, size: Size::ZERO, location: Point::ZERO }
     }
+
+    /// Creates a new [`Layout`] from order, with zero size positioned at the origin.
```
Comment (alice-i-cecile) [inline src/layout.rs:26]: It's definitely not clear what "order" does here; this needs more explanation.

File: src/flexbox.rs [hunk 2]
```diff
@@ -1800,4 +1803,52 @@ mod tests {
         assert_eq!(constants.container_size, Size::ZERO);
         assert_eq!(constants.inner_container_size, Size::ZERO);
     }
+
+    #[test]
+    fn hidden_layout_should_hide_recursively() {
+        let mut taffy = Taffy::new();
+
+        let grandchild_00 = taffy
+            .new_leaf(FlexboxLayout { display: Flex, size: Size::from_points(50.0, 50.0), ..Default::default() })
```
Comment (alice-i-cecile) [inline src/flexbox.rs:1812]: Isn't display: Flex the default? And it should always stay that way.

File: src/flexbox.rs [hunk 3]
```diff
@@ -1800,4 +1803,52 @@ mod tests {
...
+            .unwrap();
+
+        let grandchild_01 = taffy
+            .new_leaf(FlexboxLayout { display: Flex, size: Size::from_points(50.0, 50.0), ..Default::default() })
```
Comment (alice-i-cecile) [inline src/flexbox.rs:1816]: I think this text would probably be clearer if we captured this `FlexboxLayout` into a local constant.

File: src/flexbox.rs (hunk 2 above)
Comment (Weibye) [reply src/flexbox.rs:1812]: Yes `Display::Flex` is default. In the context of this test, I wanted to be explicit in showcasing that all children elements get hidden even though their Display is **not** set to `None`

File: src/flexbox.rs (hunk 3 above)
Comment (Weibye) [reply src/flexbox.rs:1816]: Good spot, didn't think of that :)

File: src/flexbox.rs (hunk 2 above)
Comment (alice-i-cecile) [reply src/flexbox.rs]: Ah I see. Okay, I'm cool with that.

File: src/layout.rs [hunk 4]
```diff
@@ -17,11 +17,23 @@ pub struct Layout {
...
+    /// Creates a new zero-[`Layout`].
+    ///
+    /// The Zero-layout has size and location set to ZERO.
+    /// Order is set to 0.
     #[must_use]
     pub const fn new() -> Self {
         Self { order: 0, size: Size::ZERO, location: Point::ZERO }
     }
+
+    /// Creates a new zero-[`Layout`] from order.
```
Comment (alice-i-cecile) [inline src/layout.rs:29]:
```suggestion
    /// Creates a new zero-[`Layout`] with the supplied `order` value.
```

File: src/layout.rs [hunk 5]
```diff
@@ -17,11 +17,23 @@ pub struct Layout {
...
+    ///
+    /// Nodes with a higher order should be rendered on top of those with a lower order.
+    /// The Zero-layout has size and location set to ZERO.
+    #[must_use]
+    pub const fn from_order(order: u32) -> Self {
```
Comment (alice-i-cecile) [inline src/layout.rs:34]:
```suggestion
    pub const fn with_order(order: u32) -> Self {
```

File: src/layout.rs [hunk 6]
```diff
@@ -17,11 +17,23 @@ pub struct Layout {
 }
 
 impl Layout {
-    /// Creates a new [`Layout`] struct with zero size positioned at the origin
+    /// Creates a new zero-[`Layout`].
+    ///
+    /// The Zero-layout has size and location set to ZERO.
+    /// Order is set to 0.
```
Comment (alice-i-cecile) [inline src/layout.rs:23]:
```suggestion
    /// The `order` value of this layout is set to the minimum value of 0.
    /// This means it should be rendered below all other [`Layout`]s.
```

File: src/layout.rs (hunk 5 above)
Comment (alice-i-cecile) [reply src/layout.rs:34]: From is typically reserved for constructors that meaningfully consume their input. In this case, it's just an integer that parameterizes the constructor.

Comment (Weibye) [reply src/layout.rs:34]: Didn't know that, but that makes sense :)

File: RELEASES.md [hunk 7]
```diff
@@ -7,7 +7,8 @@
 - Added `taffy::error::InvalidChild` Error type
 - `taffy::node::Taffy.new_leaf()` which allows the creation of new leaf-nodes without having to supply a measure function
 - Builder methods are now `const` where possible
-  - Several convenenience constants have been defined: notably `FlexboxLayout::DEFAULT`
+  - Several convenience constants have been defined: notably `FlexboxLayout::DEFAULT`
+- New `Layout` convenience constructor: `from_order(order: u32)`
```
Comment (Weibye) [inline RELEASES.md:11]:
```suggestion
- New `Layout` convenience constructor: `with_order(order: u32)`
```

**Review by alice-i-cecile — APPROVED** [review]
My suggestion broke your code, but once that's fixed this LGTM.

**Pull request merged**